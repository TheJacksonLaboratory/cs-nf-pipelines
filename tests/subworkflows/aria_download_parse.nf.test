nextflow_workflow {

    name "Test Workflow FILE_DOWNLOAD -- stock"
    script "subworkflows/aria_download_parse.nf"
    workflow "FILE_DOWNLOAD"

    test("Paired-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "PE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:1, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz', 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz'
                    ),
                    tuple(
                        'testSample_101',
                        [sampleID:'testSample_101', lane:'lane1', replicate:'NA', id:'testSample_101', size:1, ind: 'NA', sex: 'NA'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R1.fastq.gz', 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/human/hg38_WGS_sample_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
        }

    }

    test("Single-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "SE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:1, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R1.fastq.gz'
                    ),
                    tuple(
                        'testSample_101',
                        [sampleID:'testSample_101', lane:'lane1', replicate:'NA', id:'testSample_101', size:1, ind: 'NA', sex: 'NA'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/mm10_WGS_sample_R2.fastq.gz'
                    ),
                ])
                """
            }
        }

        then {
            assert workflow.success
        }

    }

    test("Multiple Lanes Paired-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "PE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:2, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R1.fastq.gz',
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R2.fastq.gz'
                    ),
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane2', replicate:'NA', id:'testSample_42', size:2, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L002_R1.fastq.gz',
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L002_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
        }

    }

    test("Multiple Lanes Single-End Data") {

        when {
            params {
                pipeline = 'wgs'
                pubdir = "tests/results"
                read_type = "SE"
            }
            workflow {
                """
                // define inputs of the workflow here. Example:
                input[0] = Channel.fromList([
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane1', replicate:'NA', id:'testSample_42', size:2, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R1.fastq.gz'
                    ),
                    tuple(
                        'testSample_42',
                        [sampleID:'testSample_42', lane:'lane2', replicate:'NA', id:'testSample_42', size:2, ind: 'Sample_42', sex: 'M'], 
                        'https://raw.githubusercontent.com/TheJacksonLaboratory/cs-nf-test/main/wgs/mouse/individual_lanes/mm10-WGS-sample_GT42-42424_TTTTTTTT-TTTTTTTT_S42_L001_R2.fastq.gz'
                    )
                ])
                """
            }
        }

        then {
            assert workflow.success
        }

    }

}
