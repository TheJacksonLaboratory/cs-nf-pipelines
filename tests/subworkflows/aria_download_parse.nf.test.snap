{
    "Single-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "mm10_WGS_sample_R2.fastq.gz:md5,aaec29ae935950d67a1cacacb6b9f4b1"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "mm10_WGS_sample_R1.fastq.gz:md5,80928ec4a7147e344ae6101321edb877"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "mm10_WGS_sample_R2.fastq.gz:md5,aaec29ae935950d67a1cacacb6b9f4b1"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "mm10_WGS_sample_R1.fastq.gz:md5,80928ec4a7147e344ae6101321edb877"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-13T10:11:11.412599565"
    },
    "Paired-End Data": {
        "content": [
            {
                "0": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "hg38_WGS_sample_R1.fastq.gz:md5,221ea2c6a58f204537970c14d4c20635",
                            "hg38_WGS_sample_R2.fastq.gz:md5,d4535a795ebc91f7782032d533067a1b"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "mm10_WGS_sample_R1.fastq.gz:md5,80928ec4a7147e344ae6101321edb877",
                            "mm10_WGS_sample_R2.fastq.gz:md5,aaec29ae935950d67a1cacacb6b9f4b1"
                        ]
                    ]
                ],
                "read_meta_ch": [
                    [
                        "testSample_101",
                        {
                            "sampleID": "testSample_101"
                        },
                        [
                            "hg38_WGS_sample_R1.fastq.gz:md5,221ea2c6a58f204537970c14d4c20635",
                            "hg38_WGS_sample_R2.fastq.gz:md5,d4535a795ebc91f7782032d533067a1b"
                        ]
                    ],
                    [
                        "testSample_42",
                        {
                            "sampleID": "testSample_42"
                        },
                        [
                            "mm10_WGS_sample_R1.fastq.gz:md5,80928ec4a7147e344ae6101321edb877",
                            "mm10_WGS_sample_R2.fastq.gz:md5,aaec29ae935950d67a1cacacb6b9f4b1"
                        ]
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-03-13T10:10:54.832949852"
    }
}